packages/wallet/src/redux/protocols/existing-channel-funding/reducer.ts

```diff
@@ -0,0 +1,349 @@
+import { SharedData, signAndStore, queueMessage, checkAndStore, getChannel } from '../../state';
+import * as states from './states';
+import { ProtocolStateWithSharedData } from '..';
+import { ExistingChannelFundingAction } from './actions';
+import * as helpers from '../reducer-helpers';
+import * as selectors from '../../selectors';
+import { proposeNewConsensus, acceptConsensus } from '../../../domain/two-player-consensus-game';
+import { theirAddress, getLastCommitment } from '../../channel-store';
+import { Commitment, nextSetupCommitment } from '../../../domain';
+import { bigNumberify } from 'ethers/utils';
+import { sendCommitmentReceived } from '../../../communication';
+import { CommitmentType } from 'fmg-core';
+import { initialize as initializeLedgerTopUp, ledgerTopUpReducer } from '../ledger-top-up/reducer';
+import { isLedgerTopUpAction } from '../ledger-top-up/actions';
+export const EXISTING_CHANNEL_FUNDING_PROTOCOL_LOCATOR = 'ExistingChannelFunding';
+export const initialize = (
```

Comment:
I guess this protocol doesn't handle postFundSetup arriving before the ledger update, which probably works fine against other wallets.